src/lib.rs

```diff
@@ -129,6 +129,60 @@ impl<'a> From<PrivatePkcs8KeyDer<'a>> for PrivateKeyDer<'a> {
     }
 }
 
+impl<'a> TryFrom<&'a [u8]> for PrivateKeyDer<'a> {
+    type Error = &'static str;
+
+    fn try_from(key: &'a [u8]) -> Result<Self, Self::Error> {
+        if key.len() <= 5 || key[0] != 0x30 {
+            return Err("Invalid key format");
+        }
+
+        // We skip the tag and the length of the sequence
+        let mut skip_len = 2;
+
+        // We skip additonal length bytes
+        if key[1] >= 0x81 {
+            skip_len += key[1] - 0x80;
+        }
+
+        match key[skip_len as usize..] {
+            [0x02, 0x01, 0x00, // Version = 0
+            0x30, 0x0D, // Sequence of two elements, which have 13 bytes length
+            0x06, 0x09, 0x2A, 0x86, 0x48, 0x86, 0xF7, 0x0D, 0x01, 0x01, 0x01, // Object Identifier for RSA Encryption (PKCS #1)
+            ..] => {
+                Ok(Self::Pkcs1(key.into()))
+            }
+
+            [0x02, 0x01, 0x01, // This is decoded as PKI header with value 1
+            ..] => {
+                Ok(Self::Sec1(key.into()))
+            }
+
+            [0x02, 0x01, 0x00, // Version = 0
+            ..] => {
+                Ok(Self::Pkcs8(key.into()))
+            }
+
+            _ => return Err("Invalid key format"),
+        }
+    }
+}
+
+#[cfg(feature = "alloc")]
+impl<'a> TryFrom<Vec<u8>> for PrivateKeyDer<'a> {
+    type Error = &'static str;
+
+    fn try_from(key: Vec<u8>) -> Result<Self, Self::Error> {
+        let private_key_der = PrivateKeyDer::try_from(&key[..])?;
```

Comment:
Nit: I think it would be okay to just inline the binding here? (As in, `match PrivateKeyDer::try_from(&key[..])? { .. }`.)